src/checks/cargo/run-cargo.ts

```diff
@@ -0,0 +1,18 @@
+import { ExitInformation } from '../../unix'
+import { runJsonLinesCommand } from '../checks-common'
+import { getEnvRustFlags, touchRustFiles } from './cargo'
+import { CargoMessage } from './cargo-types'
+
+export async function runCargo(args: string[], ci = true): Promise<[ExitInformation, CargoMessage[]]> {
+  // Description for why this is needed, can be found
+  // at the `touchRustFiles` definition
+  await touchRustFiles()
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  return runJsonLinesCommand<CargoMessage>('cargo', args, {
+    env: {
+      RUSTFLAGS: getEnvRustFlags(ci),
+      ...process.env
```

Comment:
You might want to change the order here, if the RUSTFLAGS is set in the environment it will get overwritten when the object is merged, so the trimming you are doing getEnvRustFlags will be null and void.

Reply:
Hmm, maybe I didn't test it thoroughly then, or maybe I'm just confusing it with Rust :)